breacharbiter.go

```diff
@@ -472,92 +420,146 @@ func (b *breachArbiter) breachObserver(contract *lnwallet.LightningChannel,
 		// commitment transaction. This output is just a regular p2wkh
 		// output.
 		localSignDesc := breachInfo.LocalOutputSignDesc
-		localWitness := func(tx *wire.MsgTx, hc *txscript.TxSigHashes,
-			inputIndex int) ([][]byte, error) {
-
-			desc := localSignDesc
-			desc.SigHashes = hc
-			desc.InputIndex = inputIndex
-
-			return lnwallet.CommitSpendNoDelay(b.wallet.Signer, desc, tx)
-		}
+		localWitnessType := lnwallet.CommitmentNoDelay
 
 		// Next we create the witness generation function that will be
 		// used to sweep the cheating counterparty's output by taking
 		// advantage of the revocation clause within the output's
 		// witness script.
 		remoteSignDesc := breachInfo.RemoteOutputSignDesc
-		remoteWitness := func(tx *wire.MsgTx, hc *txscript.TxSigHashes,
-			inputIndex int) ([][]byte, error) {
+		remoteWitnessType := lnwallet.CommitmentRevoke
 
-			desc := breachInfo.RemoteOutputSignDesc
-			desc.SigHashes = hc
-			desc.InputIndex = inputIndex
-
-			return lnwallet.CommitSpendRevoke(b.wallet.Signer, desc, tx)
-		}
-
-		// Finally, with the two witness generation funcs created, we
-		// send the retribution information to the utxo nursery.
+		// Finally, we send the retribution information into the breachArbiter
+		// event loop to deal swift justice.
 		// TODO(roasbeef): populate htlc breaches
-		b.breachedContracts <- &retributionInfo{
+		b.breachedContracts <- &retribution{
 			commitHash: breachInfo.BreachTransaction.TxHash(),
 			chanPoint:  *chanPoint,
 
 			selfOutput: &breachedOutput{
-				amt:         btcutil.Amount(localSignDesc.Output.Value),
-				outpoint:    breachInfo.LocalOutpoint,
-				witnessFunc: localWitness,
+				amt:            btcutil.Amount(localSignDesc.Output.Value),
+				outpoint:       breachInfo.LocalOutpoint,
+				signDescriptor: localSignDesc,
+				witnessType:    localWitnessType,
 			},
 
 			revokedOutput: &breachedOutput{
-				amt:         btcutil.Amount(remoteSignDesc.Output.Value),
-				outpoint:    breachInfo.RemoteOutpoint,
-				witnessFunc: remoteWitness,
+				amt:            btcutil.Amount(remoteSignDesc.Output.Value),
+				outpoint:       breachInfo.RemoteOutpoint,
+				signDescriptor: remoteSignDesc,
+				witnessType:    remoteWitnessType,
 			},
 
-			doneChan: make(chan struct{}),
+			htlcOutputs: []*breachedOutput{},
```

Comment:
The `doneChan` should still be initialized. Also there's no reason to populate the `htlcOutputs` field as it isn't being used yet.